### PR TITLE
Trees and flora

### PR DESCRIPTION
**Pull request by Delete69**
## About the PR 
Adds tree entities and flora decals. This is my first meaty PR, so any feedback on how I did things is welcome. I tried to understand and tweak as much as I could before I pushed it. As they stand, the trees are not constructible, indestructible, and only really intended for mapping usage to add flavor.

**Screenshots**

decals
trees

**Changelog**

:cl: Delete
- add: Added trees and flora for mappers.

### CODE REVIEW

**Comment by Delete69** [comment]
yeah idk what a ClickTestRotatingCornerVisible is

**Comment by metalgearsloth** [comment]
> yeah idk what a ClickTestRotatingCornerVisible is

You used the wrong separators

File: Resources/Prototypes/Entities/Objects/Decoration/flora.yml [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+    bodyType: Static
+  - type: Fixtures
+    fixtures:
+    - shape:
+        !type:PhysShapeAabb
+          bounds: "-0.1,-1.5,0.1,-0.8"
+      mass: 500
+      layer:
+      - Opaque
+      - Impassable
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Decoration/flora.yml:83]: Bunch duplicated with BaseTree?

File: Resources/Prototypes/Entities/Objects/Decoration/flora.yml [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+    sprite: Objects\Decoration\Flora\flora_trees.rsi
+    netsync: false
+    drawdepth: Overdoors
+  - type: Physics
+    bodyType: Static
+  - type: Fixtures
+    fixtures:
+    - shape:
+        !type:PhysShapeAabb
+          bounds: "-0.35,-1.3,0.35,-0.5"
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Decoration/flora.yml:16]: These should really be circle.

File: Resources/Prototypes/Entities/Objects/Decoration/flora.yml [hunk 3]
```diff
@@ -0,0 +1,292 @@
+- type: entity
+  id: BaseTree
+  abstract: true
+  components:
+  - type: Clickable
+  - type: Sprite
+    sprite: Objects\Decoration\Flora\flora_trees.rsi
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Decoration/flora.yml:7]: Should have a placementmode to default to tile centre even if it's not anchored.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by Delete69** [comment]
> > yeah idk what a ClickTestRotatingCornerVisible is
> 
> You used the wrong separators

I really don't see where anything is wrong, I can't decipher the errors and my eye is untrained.

> Bunch duplicated with BaseTree?

I guess I'll solve this with parenting a parent?

> These should really be circle.

I started out with a circle but it wasn't as good as. Using a rectangle allowed the player sprite to get up close to the trunk, while still being able to hide behind the foliage and not have their head get covered by the trunk of the trees. Each type of tree sprite required a different shape bounds. Is a circle really something you require here?

image

> Should have a placementmode to default to tile centre even if it's not anchored.

I've used these trees in mapping for 2 weeks now, and 9 times out of 10 I wanted to PlaceFree so I removed SnapgridCenter from my yaml. Is this really required as well?

**Comment by metalgearsloth** [comment]
> I really don't see where anything is wrong, I can't decipher the errors and my eye is untrained.

You used \ instead of /

> Each type of tree sprite required a different shape bounds. Is a circle really something you require here?

The bounds look like they should be wider than they are tall to me if you want to make hiding behind them more of a thing.

> I've used these trees in mapping for 2 weeks now, and 9 times out of 10 I wanted to PlaceFree so I removed SnapgridCenter from my yaml. Is this really required as well?

It's not a huge deal but as long as the tree sprite is offset then the center should be the base.

**Comment by Delete69** [comment]
I fixed up the separators, tidied up the parents. I left the bounds like they are because they are tweaked to perfection and work really keen. Theres a fine line between being able to hide behind the tree due to drawdepth and not having you're head disappear under the trunk when you stand below it. I also left out the SnapgridCenter because I believe its the right way to go for mappers.

**Review by mirrorcult — APPROVED** [review]
we really need decal placement sorting by tags now tbh

**Pull request merged**